File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int h;
    do
    {
        h = get_int("Choose a height for the pyramids among integers from 0 to 23: ");
    }

    while (h < 0 || h > 23);

    for (int i = 0; i < h; i++)
    {

        //print h-1-i spaces for the left
        printf("%*s", h - 1 - i, "");
        //print i+1 hashes for left
        for (int a = 0; a < i + 1; a++)
        {
            printf("#");
        }

        //print 2 spaces for gap
        printf("  ");

        //print i+1 hashes for right
        for (int a = 0; a <= i; a++)
        {
            printf("#");
        }

        printf("\n");
    }

}

File: mario.py
from cs50 import get_int

# prompt user for height, keep reprompting if height is invalid
while True:
    h = get_int("Choose a height for the pyramids among integers from 0 to 23: ")
    if h >= 0 and h <= 23:
        break

# iterate by row
for i in range(h):
    # print h - 1 spaces for the left
    print((h - 1 - i) * " ", end='')
    # print i + 1 hashes for left
    print((i + 1) * "#", end='')

    # print 2 spaces for gap
    print("  ", end='')

    # print i + 1 hashes for right
    print((i + 1) * "#", end='')

    print()
